src/Microsoft.AspNet.Authentication/DataHandler/PropertiesSerializer.cs

```diff
@@ -39,26 +41,29 @@ public AuthenticationProperties Deserialize(byte[] data)
             }
         }
 
-        public static void Write([NotNull] BinaryWriter writer, [NotNull] AuthenticationProperties properties)
+        public virtual void Write([NotNull] BinaryWriter writer, [NotNull] AuthenticationProperties properties)
         {
             writer.Write(FormatVersion);
             writer.Write(properties.Items.Count);
-            foreach (var kv in properties.Items)
+
+            foreach (var item in properties.Items)
             {
-                writer.Write(kv.Key);
-                writer.Write(kv.Value);
+                writer.Write(item.Key ?? string.Empty);
```

Comment:
Fixes the bug seen in https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/issues/126.